### DISCUSSION TITLE
How to model multiple supervisor cards but where only one is active

### DISCUSSION THREAD

**Discussion by fknorn** [netbox-community/netbox#17863] [Q&A]
Hi there,

I'm curious how you would solve this. I'm trying to model a Cisco N7k switch [community model](https://github.com/netbox-community/devicetype-library/blob/master/device-types/Cisco/N77-C7706.yaml). These are typically operated with two supervisor cards [the module in the community database](https://github.com/netbox-community/devicetype-library/blob/master/module-types/Cisco/N77-SUP2E.yml), which are inserted into the module slots (in the real world and in netbox).

Now, each of these supervisor cards has a `console` and a `mgmt0` port. But only one supervisor card is active and "visible" on the CLI. By that I mean that `show int brief` will only list _one_ `mgmt0` interface. The inactive supervisor's interface, in other words, doesn't show up.

However, it physically exists and can (is) patched to some management switch. Whenever the primary supervisor card fails, the switch just changes over and the other card (together with its management and console-connections) become active.

So how should this be modeled?

Thanks!

**Comment by candlerb** [comment]
From the point of cabling, you need two physical ports, which will need different names. But from the point of configuration, there's just one.

You could create two physical ports (e.g. "mgmt0-pri", "mgmt0-sec") and one virtual port ("mgmt0"), and use mgmt0 for the source-of-truth for interface configuration. But you don't get any association between the physical ports and the logic interface.

You could make "mgmt0" be a bridge, and set mgmt0-pri and mgmt0-sec to point to it with the "bridge" attribute. That could be a reasonable option, even though the two interfaces are not *actually* bridged together.

Or for simplicity you could take the asymmetric approach: have "mgmt0" on the primary card where you configure the IP address, "mgmt0-sec" on the secondary card used purely for cable termination, and ignore the possibility that it could failover to the secondary.
